runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -519,6 +519,7 @@ private void cleanUpForRetry() {
 
         // Re ask for the Head, if it changes while we were trying.
         findAndSetLogHead();
+        findAndSetLogTail();
```

Comment:
Does the loadMaps keep on failing if it spans a trim, or does it just load partial checkpoints?

Reply:
It will catch the exception for a few time and retry. What we saw in the bug is that only partial number of keys are loaded, the old checkpoint is trimmed as the head is advanced, and the new checkpoints are not in the scope of the old tail.

Reply:
I see.

If only the head was being updated on a trimmed exception wouldn't that mean on the next retry the tail will be less than the head and some of the range should still be trimmed?

I think we should explicitly fail (which doesn't seem to be the case) instead of succeeded. In other words, If loadmaps always spans a checkpoint/trim cycle, the problem will still happen, it will silently fail.